packages/xcoobee-cookie-kit-react/src/CookieKitContainer.js

```diff
@@ -206,21 +206,25 @@ export default class CookieKitContainer extends React.PureComponent {
     const { campaignReference, detectCountry } = this.props;
     const { countryCode } = this.state;
 
+    const defaultCountryCode = "EU";
+
     let promise = null;
 
     if (countryCode) {
       return Promise.resolve(countryCode);
     }
-    if (!!campaignReference && detectCountry) {
-      promise = Promise.resolve(fetchCountryCodeForSubscribers(campaignReference));
+    if (detectCountry) {
```

Comment:
To make the code easier to follow, let's move the `let promise = null;` into this `if` statement. Then let's move the `return promise\n      .catch(...` after the following `if/else` statement. Then we can get rid of the `else` clause of this `if` statement and simply `return Promise.resolve(defaultCountryCode);`. This way, the `promise.catch(...).then(...);` is grouped together with the corresponding code branch.

So, something like:

```suggestion
    if (detectCountry) {
      let promise;
      if (campaignReference) {
        promise = Promise.resolve(fetchCountryCodeForSubscribers(campaignReference));
      } else {
        promise = Promise.resolve(fetchCountryCode());
      }
      return promise
        .catch((error) => {
          console.error(error);
          this.setState({ countryCode: defaultCountryCode });

          return defaultCountryCode;
        })
        .then((cCode) => {
          saveCountryCode(cCode);
          this.setState({ countryCode: cCode });
          return cCode;
        });
    }
    return Promise.resolve(defaultCountryCode);
```